test/controllers/PenaltyDetailsController.test.ts

```diff
@@ -1,55 +1,44 @@
-import 'reflect-metadata'
-import { createApplication } from '../ApplicationFactory';
-import * as request from 'supertest'
-import { createSubstituteOf } from '../SubstituteFactory';
+import 'reflect-metadata';
 
+import { createApp, getDefaultConfig } from '../ApplicationFactory';
+import * as request from 'supertest';
 import '../../src/controllers/PenaltyDetailsController';
-import { RedisService } from '../../src/services/RedisService';
 import { MOVED_TEMPORARILY, OK, UNPROCESSABLE_ENTITY } from 'http-status-codes';
 import { expect } from 'chai';
 import { PenaltyIdentifier } from '../../src/models/PenaltyIdentifier';
-import { PENALTY_DETAILS_PAGE_URI, OTHER_REASON_DISCLAIMER_PAGE_URI } from '../../src/utils/Paths'
+import { PENALTY_DETAILS_PAGE_URI, OTHER_REASON_DISCLAIMER_PAGE_URI } from '../../src/utils/Paths';
+import { createFakeSession } from '../utils/session/FakeSessionFactory';
+import { sessionKey } from '../../src/controllers/PenaltyDetailsController';
 
 const pageHeading = 'What are the penalty details?';
 const errorSummaryHeading = 'There is a problem with the information you entered';
 
-const app = createApplication(container => {
-    container.bind(RedisService).toConstantValue(createSubstituteOf<RedisService>());
-});
+const config = getDefaultConfig();
 
 describe('PenaltyDetailsController', () => {
     describe('GET request', () => {
-
-        it('should return 200 when trying to access page without a session', async () => {
-            await request(app).get(PENALTY_DETAILS_PAGE_URI)
-                .expect(response => {
-                    expect(response.status).to.be.equal(OK);
-                    expect(response.text).to.contain(pageHeading)
-                        .and.not.contain(errorSummaryHeading);
-                })
+        it('should return 302 when trying to access page without a session', async () => {
+            await request(createApp()).get(PENALTY_DETAILS_PAGE_URI)
+                .expect(MOVED_TEMPORARILY);
```

Comment:
This might not be enough. Check where user agent was moved to.

Reply:
Same sort of test should be applied to other controllers protected by auth middleware. Could be next PR if we decided to keep this PR small. If so this test could be moved to separate PR.

Reply:
We have removed this test and in another PR we can address more tests regarding auth middleware